src/Functions/PartitionByHyperplanes.cpp

```diff
@@ -0,0 +1,338 @@
+        size_t coordinate_index)
+    {
+        for (size_t nt = vectors_data_index; nt < vectors_data_index + tile_size && nt < input_rows_count; ++nt)
+            for (size_t mt = normals_data_index; mt < normals_data_index + tile_size && mt < input_rows_count; ++mt)
+                for (size_t kt = coordinate_index; kt < coordinate_index + tile_size && kt < dimension; ++kt)
+                {
+                    auto vector_element = nested_vectors_data.getFloat32(nt * dimension + kt);
+                    auto normal_element = nested_normals_data.getFloat32(mt * dimension + kt);
+                    auto offset_element = nested_offsets_data->getFloat32(mt * dimension + kt);
+
+                    col_res.getElement(nt * input_rows_count + mt) += (vector_element - offset_element) * normal_element;
+                }
+    }
+
+#if defined(__AMX_BF16__)
+    template <class ColumnType>
+    static void subVectorOffsetTile(
+        const ColumnType & nested_vectors_data,
+        const IColumn * nested_offsets_data,
+        size_t input_rows_count,
+        size_t dimension,
+        size_t vectors_data_index,
+        size_t offsets_data_index,
+        size_t coordinate_index)
+    {
+        for (size_t i = 0; i < tile_size && vectors_data_index + i < input_rows_count && offsets_data_index + i < input_rows_count; ++i)
+            for (size_t j = 0; j < tile_size && coordinate_index + j < dimension; ++j)
+            {
+                auto vector_element = nested_vectors_data.getFloat64((vectors_data_index + i) * dimension + coordinate_index + j);
+                auto offset_element = nested_offsets_data->getFloat64((offsets_data_index + i) * dimension + coordinate_index + j);
+
+                temp_tile[i * tile_size + j] = vector_element - offset_element;
+            }
+    }
+
+    template <class ColumnType>
+    static void amxTileMultiply()
+    {
+        if constexpr (std::is_same_v<ColumnType, ColumnFloat32>)
+            _tile_dpbf16ps(0, 1, 2);
+        else if (std::is_same_v<ColumnType, ColumnInt8>)
+            _tile_dpbssd(0, 1, 2);
+    }
+
+    struct TileConfig
+    {
+        uint8_t palette_id;
+        uint8_t start_row;
+        uint8_t reserved_0[14];
+        uint16_t colsb[16]; 
+        uint8_t rows[16]; 
+    };
+
+    template <class ColumnType, class ResultColumnType>
+    static void multiplyTileAMX(
+        const ColumnType & nested_vectors_data,
+        const ColumnType & nested_normals_data,
+        const IColumn * nested_offsets_data,
+        ResultColumnType & col_res,
+        size_t input_rows_count,
+        size_t dimension,
+        size_t vectors_data_index,
+        size_t normals_data_index,
+        size_t coordinate_index)
+    {
+        subVectorOffsetTile(
+            nested_vectors_data,
+            nested_offsets_data,
+            input_rows_count,
+            dimension,
+            vectors_data_index,
+            normals_data_index);
+
+        TileConfig tileinfo{
```

Comment:
Comment with general description what is the tile and how algorithm is executed may be helpful